src/components/UserSelector.tsx

```diff
@@ -1,35 +1,96 @@
-import React from 'react';
+import classNames from 'classnames';
+import { useEffect, useState } from 'react';
+import { User } from '../types/User';
+import { apiActions } from '../utils/apiAction';
+import { ErrorNotification } from '../types/ErrorNotification';
+import { Post } from '../types/Post';
+
+type Props = {
+  userSelected: User | null,
+  setUserSelected: (user: User) => void;
+  setErrorNotification: (error: (prevError: ErrorNotification)
+  => ErrorNotification) => void;
+  setSelectedPost?: (posts: Post) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  userSelected,
+  setUserSelected,
+  setErrorNotification,
+}) => {
+  const [isSelected, setIsSelected] = useState(false);
+  const [users, setUsers] = useState<User[] | null>(null);
+
+  const handleClickSelect = () => {
+    setIsSelected(!isSelected);
+  };
+
+  useEffect(() => {
+    apiActions.getAllUsers()
+      .then(setUsers)
+      .catch(() => {
+        setErrorNotification((error) => ({ ...error, users: true }));
+      });
+  }, []);
```

Comment:
Same
![2023-10-18_15h51_32](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/61904995/d08f7790-95b9-4fa2-82b5-16347b1ac064)
Please read about dependency array
https://react.dev/reference/react/useEffect#useeffect